Enterprise-CMCS/managed-care-review
# MCR-4244 User can submit linked rate with deprecated programs, no errors

## Summary
Follow up to #2547 when user could not submit and also error was still showing in some places

This PR includes:
- reimplementation of the frontend work - calculating if the rate is linked more directly
- reimplementation of backend validation- account for linked rates in `parseAndSubmit` (have to shoehorn this because actually this legacy API actually isn't supposed to handle linked rates at all and has no concept of linking in its types) 

#### Related issues
https://jiraent.cms.gov/browse/MCR-4244

#### Screenshots
- renders the deprecated rate programs when no rate programs present on the last submitted version of rate shown to state user

## QA guidance

1. Log in with `toph2@example.com` user on review app. This is a user in a state that has historic rate submissions. Create a new submission
2. Select "contract action and rate certification" and fill out form
3. On the Rate details page link a rate certification that was submitted prior to when the rate name enhancement feature went live.
4. Click continue and fill out the rest of the submission
5. When you arrive at the Review and submit page you should see old deprecated rate programs and no cilentside missing data errors
6. You can submit that package with no backend errors
